test/common/stats/symbol_table_fuzz_test.cc

```diff
@@ -0,0 +1,43 @@
+#include "common/common/assert.h"
+#include "common/common/base64.h"
+#include "common/stats/symbol_table_impl.h"
+
+#include "test/fuzz/fuzz_runner.h"
+
+namespace Envoy {
+namespace Stats {
+namespace Fuzz {
+
+// Fuzzer for symbol tables.
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  FuzzedDataProvider provider(buf, len);
+  SymbolTableImpl symbol_table;
+  StatNamePool pool(symbol_table);
+  bool provider_empty = provider.remaining_bytes() == 0;
+
+  const uint64_t ResetPoolAfterN = 10000;
```

Comment:
Ideally, we tune this to be as low as possible while still covering interesting range of behaviors. If there is going to be very unlikely difference between 1k and 10k, then go for 1k.

Reply:
No idea what the coverage is yet. How do I find that out?

Reply:
I'd start with intuition. Are there any limits in the system that would indicate that you need an input size at least N to fully explore useful behavior. An example is a buffer fuzz test. If your buffers internally chunk at 1KB, then you definitely want your test data to be > 1KB to explore edge cases. But there is no point doing 100MB.

I would suggest dropping this to something like 100 or less for performance reasons. Fuzzer performance is almost as important as the fuzzer coverage in a single run, since the fuzzing engine can effectively explore the state space when it is fast.

Reply:
I simplified and dropped the memory bounds reset, as it doesn't appear to be needed. I had put it in because the memory usage looked high, but based on debug prints it never took effect anyway.